### PR TITLE
Add Python 3.8 Test

### PR DESCRIPTION
**Pull request by fengzhou-msft**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -408,7 +413,7 @@ jobs:
       REDUCE_SDK: 'True'
       AZURE_CLI_TEST_TARGET_PROFILE: '2019-03-01'
 
-- job: RunAutomationReduced20180301
+- job: RunAutomation20180301
```
Comment (Juliehzl) [inline azure-pipelines.yml]: Why do you run reduced tests for profile 2019-03-01 but 2018-03-01 not? What is the difference?

Comment (fengzhou-msft) [reply azure-pipelines.yml]: This is the current test settings for python 3.6. I modified the name to reflect the settings.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -434,6 +439,86 @@ jobs:
...
+    inputs:
+      TargetPath: '$(Build.ArtifactStagingDirectory)/pypi'
+      artifactName: pypi
+  - bash: |
+      pip install --force-reinstall wheel
+  - task: Bash@3
+    displayName: 'Run Test Automation'
+    inputs:
+      targetType: 'filePath'
+      filePath: ./scripts/ci/test_automation.sh
```
Comment (Juliehzl) [inline azure-pipelines.yml]: Actually I also have a question for test_automation.sh file. Typically we run test based on `azdev test`, but in ADO CI system, tests are run based on python https://github.com/Azure/azure-cli/blob/dev/scripts/ci/test_automation.sh#L57.

**Comment by Juliehzl** [comment]
LGTM in general and leave some questions.

File: azure-pipelines.yml (hunk 1 above)
Comment (Juliehzl) [reply azure-pipelines.yml]: I see. Just out of curiosity, ReducedAutomation is run when REDUCE_SDK is set to true.  But through the [scripts](https://github.com/Azure/azure-cli/blob/dev/scripts/ci/test_automation.sh#L35), I cannot see anything special besides reduced modules.

Comment (fengzhou-msft) [reply azure-pipelines.yml]: Not familiar with testing details, add @haroldrandom for digging in.

File: azure-pipelines.yml (hunk 2 above)
Comment (fengzhou-msft) [reply azure-pipelines.yml]: Another good question to look into, this could be a discrepancy in developing and ci testing, @haroldrandom

Comment (haroldrandom) [reply azure-pipelines.yml]: Yeah. It's a TODO for CI

**Pull request merged**